modules/diskq/logqueue-disk-non-reliable.c

```diff
@@ -383,10 +383,22 @@ _peek_head(LogQueue *s)
   LogQueueDiskNonReliable *self = (LogQueueDiskNonReliable *)s;
   LogMessage *msg = NULL;
 
-  g_mutex_lock(&s->lock);
+  if (self->front_cache_size != 0)
+    {
+      if (_peek_from_memory_queue_head(&self->front_cache_output, &msg))
+        return msg;
 
-  if (_peek_from_memory_queue_head(&self->front_cache, &msg))
-    goto success;
+      g_mutex_lock(&s->lock);
+
+      _maybe_move_messages_among_queue_segments(self);
+
+      g_mutex_unlock(&s->lock);
+
+      if (_peek_from_memory_queue_head(&self->front_cache_output, &msg))
+        return msg;
+    }
+
+  g_mutex_lock(&s->lock);
```

Comment:
I think a peek operation is expected to run quickly. `_maybe_move_messages_among_queue_segments()` might even do some disk IO operations, which can be quite slow. I am a bit hesitant to make this modification.

Let's see where it is used:
This function will be called by `log_queue_peek_head()`, which is only used in `logthrdestdrv.c` in `_perform_inserts()`. Short after its call, `log_queue_pop_head()` will be called, which does another round of moving anyways, so I think it is unnecessary to do it twice, and it is not necessary for peeking, so I would remove it from there.

This commit seems deliberate, so maybe I am wrong. Am I missing something?